### PR TITLE
HTML sanitizer - clarify the allow + disallow case

### PR DESCRIPTION
**Pull request by hamishwillee**
The HTML sanitizer docs did not specify what happens if the [config](https://developer.mozilla.org/en-US/docs/Web/API/Sanitizer/Sanitizer#parameters) uses both an allow and a drop list for elements/attributes at the same time. 

This clarifies that it is not recommended and shows what would happen.

This follows on from https://github.com/mdn/content/pull/18264 and the discussion in https://github.com/WICG/sanitizer-api/pull/166#issuecomment-1194825415

### CODE REVIEW

File: files/en-us/web/api/sanitizer/sanitizer/index.md [hunk 1]
```diff
@@ -31,21 +31,33 @@ new Sanitizer(config)
 
     - `allowElements` {{optional_inline}}
       - : An {{jsxref('Array')}} of {{jsxref('String','strings')}} indicating elements that the sanitizer should not remove.
+        All elements not in the array will be dropped.
```
Comment (hamishwillee) [inline files/en-us/web/api/sanitizer/sanitizer/index.md:34]: Note:
1. This is the obvious consequence of the statement above, but it is important so saying it again. 
   Note that I don't that the "block/drop" cases don't remove all other elements by choice.

File: files/en-us/web/api/sanitizer/sanitizer/index.md [hunk 2]
```diff
@@ -31,21 +31,33 @@ new Sanitizer(config)
...
       - : A {{jsxref('Boolean')}} value set to `false` (default) to remove HTML comments.
         Set to `true` in order to keep comments.
 
+> **Note:** `allowElements` creates a sanitizer that will drop any elements that are not in `allowElements`, while `blockElements` and `dropElements` create a sanitizer that will allow all elements except those in these properties.
+>
+> `blockElements` and `dropElements` are processed before `allowElements`.
+> If you specify both properties, the elements in `blockElements` or `dropElements` will be discarded first, followed by any elements not in `allowElements`.
+> So while it is possible to specify both types of properties at the same time, the intent can always be more clearly captured using just one type.
+>
+> The same applies to `allowAttributes` and `dropAttributes`.
```
Comment (hamishwillee) [inline files/en-us/web/api/sanitizer/sanitizer/index.md:59]: FYI This is the important bit!

File: files/en-us/web/api/sanitizer/sanitizer/index.md [hunk 3]
```diff
@@ -65,7 +77,7 @@ const sanitized =  new Sanitizer().sanitizeFor("div", unsanitized);
 // Result (innerHTML of 'sanitized'): script will be removed: "abc alert(1) def"
 ```
 
-<!-- Add other examples showing use of parameter when it is implemented somewhere -->
+<!-- Add other examples showing use of parameter -->
```
Comment (hamishwillee) [inline files/en-us/web/api/sanitizer/sanitizer/index.md:80]: We should have more examples for the new cases, both here and in the top level doc: https://developer.mozilla.org/en-US/docs/Web/API/HTML_Sanitizer_API

@najmiehsa would love you to do examples in a separate PR if you would like. Probably not covering the "use both block and allow" case though.

Comment (najmiehsa) [reply files/en-us/web/api/sanitizer/sanitizer/index.md:80]: @hamishwillee Sure I do. Thank you.

**Review by sideshowbarker — APPROVED** [review]

**Comment by najmiehsa** [comment]
@hamishwillee I add examples for the sub-document. If this is alright then I can add examples to the https://developer.mozilla.org/en-US/docs/Web/API/HTML_Sanitizer_API as well. 
Please let me know.

**Comment by hamishwillee** [comment]
Thanks @najmiehsa Let's see how the examples go, and then we can think about what needs to go in the top level document. It might be that we just link to them instead.
